### PR TITLE
Create View for creating musig addresses in multisig-toolkit

### PR DESCRIPTION
**Pull request by jnaulty**
## Description 

Add a tab to the multisig-toolkit for creating Sui MultiSig Addresses 'on-the-fly' ✈️  
Ref #12932

## Test Plan 

* Get PubKeys!
```
  AB3VXSdvFq6ym0ulqnVNuEI9xxI5b8RY21azMU6ybWp1
  ADue3nTTYz9pE75pMKEnnLIrcTSS/c0uwr6WEU5nz4RM
  AIPVXOb2rbm8dX0LBL4PEgN4HPQDpwUIHpIy2THr/it1
```

* run locally `pnpm run dev`
* goto [localhost:5173/multisig-address](http://localhost:5173/multisig-address)
* add pubkeys and weights
* create!
---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [ ] protocol change
- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

**Comment by jnaulty** [comment]
Current Progress:

* can input N pubkeys + weights 💪 
* hardcoded threshold 🙀 
* output is currently to console 📜 

image

**Comment by jnaulty** [comment]
Current Progress (pairing with @georgedigkas )
image

**Comment by jnaulty** [comment]
Current Status
image

**Review by georgedigkas — APPROVED** [review]

File: dapps/multisig-toolkit/src/routes/multisig-address.tsx [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+		},
+	});
+	const { fields, append, remove } = useFieldArray({
+		control,
+		name: 'pubKeys',
+	});
+
+	// Perform generation of multisig address
+	const onSubmit = (data: FieldValues) => {
+		console.log('data', data);
```
Comment (Jordan-Mysten) [inline dapps/multisig-toolkit/src/routes/multisig-address.tsx:40]: @jnaulty Console log 😨

File: dapps/multisig-toolkit/src/routes/multisig-address.tsx [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+
+		let pks: PubkeyWeightPair[] = [];
+		data.pubKeys.forEach((item: any) => {
+			console.log(item.pubKey);
```
Comment (Jordan-Mysten) [inline dapps/multisig-toolkit/src/routes/multisig-address.tsx:44]: We really are logging everywhere...

File: dapps/multisig-toolkit/src/routes/multisig-address.tsx [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+			const pkBytes = fromB64(item.pubKey);
+			const flag: number = pkBytes[0];
+			console.log(flag);
```
Comment (Jordan-Mysten) [inline dapps/multisig-toolkit/src/routes/multisig-address.tsx:47]: And when I say everywhere...

File: dapps/multisig-toolkit/src/routes/multisig-address.tsx [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+			const rawPkBytes = toB64(pkBytes.slice(1));
+			const schemeFlag = (SIGNATURE_FLAG_TO_SCHEME as { [key: number]: string })[flag];
+			const scheme = schemeFlag as SignatureScheme;
+
+			const pk = publicKeyFromSerialized(scheme, rawPkBytes);
+			console.log(pk);
```
Comment (Jordan-Mysten) [inline dapps/multisig-toolkit/src/routes/multisig-address.tsx:53]: I really do mean...

File: dapps/multisig-toolkit/src/routes/multisig-address.tsx [hunk 5]
```diff
@@ -0,0 +1,196 @@
...
+			pks.push({ pubKey: pk, weight: item.weight });
+		});
+		console.log('pks:', pks);
```
Comment (Jordan-Mysten) [inline dapps/multisig-toolkit/src/routes/multisig-address.tsx:56]: Every...

File: dapps/multisig-toolkit/src/routes/multisig-address.tsx [hunk 6]
```diff
@@ -0,0 +1,196 @@
...
+		const multisigSuiAddress = toMultiSigAddress(pks, data.threshold);
+		console.log('multisigSuiAddress', multisigSuiAddress);
```
Comment (Jordan-Mysten) [inline dapps/multisig-toolkit/src/routes/multisig-address.tsx:58]: ...where

**Pull request merged**